File: Recursividade/Power.java
package Recursividade;

public class Power {
    public static int power(int a, int b) {

        if(b == 0) return 1;

        return tailPower(a, b, 1);
    }

    public static int tailPower(int a, int b , int c) {

        if( b == 0) return c;

        return tailPower(a, b-1, c*a);
    }

    public static void main(String[] args) {
        System.out.println(power(3, 2));
        System.out.println(power(3, 3));
        System.out.println(power(2, 10));
        System.out.println(power(10, 4));
    }
}

File: Recursividade/python/power.py
import unittest

def exponentiation(base:int, power:int, acc:int = 1) -> int:
    if power < 0:
        return 0

    if power == 0:
        return acc

    return exponentiation(base, power - 1, acc * base)

class Tester(unittest.TestCase):

    def test_power_should_return_when_positive_power(self,):
        expected = 1024
        got = exponentiation(2, 10)
        self.assertEqual(expected, got)

    def test_zero_power_should_return_1(self,):
        self.assertEqual(1, exponentiation(50, 0))
        self.assertEqual(1, exponentiation(-50, 0))